packages/backend-app-api/src/config/config.ts

```diff
@@ -39,10 +39,13 @@ export async function createConfigSecretEnumerator(options: {
 }): Promise<(config: Config) => Iterable<string>> {
   const { logger, dir = process.cwd() } = options;
   const { packages } = await getPackages(dir);
+
   const schema =
     options.schema ??
     (await loadConfigSchema({
-      dependencies: packages.map(p => p.packageJson.name),
+      dependencies: packages
+        .filter(p => p.dir === dir)
```

Comment:
Hmm, I don't think this is a safe assumption, cwd might be the workspace root as well. The backend could be started with for example either `cd packages/backend; node .` or `node packages/backend`. We'd really want to figure out exactly which package we're runnig, maybe through `process.argv[1]`?

Reply:
Tried to give it a go, but [this line](https://github.com/aramissennyeydd/backstage/blob/0f2242a32cfa992efcc885ee15bc382695f13089/packages/config-loader/src/schema/collect.ts#L68-L73) is giving me a bunch of trouble in tests, any ideas on an alternative route?

Reply:
This might have some useful bits for mocking module resolution: https://github.com/backstage/backstage/blob/cf59a34de62b9ee230093e61ac0e39e5609264f7/packages/config-loader/src/schema/collect.test.ts

Basically setting up a module structure in the external dir and switching cwd if needed.

Reply:
That definitely helped, I think I'm mostly there, but the directory resolving seems weird to me still. I had to hack my way around the case where the current directory is `a` and we're trying to resolve `a` because that is the package we start collecting schemas from. `require.resolve` seems to start searching in `node_modules` before checking the current package, leading to it throwing even though it's in the correct directory. to get around it, I just pass in the direct dependencies from the correct package, but that seems like another not quite correct assumption.

Reply:
Hmm, how do you mean? That's only the frontend config, or do you mean to do something similar for the backend?

Reply:
Was thinking something similar for the backend, collecting it before we go into this so it's easier to reason about and not as dynamic.

Reply:
Oh btw to clarify: `require` and `require.resolve` does not resolve packages in the same directly by name, you'd need a file path for that. Only modules in `node_modules` are resolved by name: https://nodejs.org/api/modules.html#loading-from-node_modules-folders

Reply:
Ah, interesting - yeah that's the case I was running into. I think my dependency grabbing gets rid of that case, but it seems a little excessive so open to alternatives.